src/nitbuilder.nit

```diff
@@ -0,0 +1,286 @@
+		if visibility == null then visibility = public_visibility
+		var mproperty = toolcontext.modelbuilder.try_get_mproperty_by_name(null, mclassdef, name).as(nullable MMethod)
+		if mproperty == null then mproperty = new MMethod(mclassdef, name, mclassdef.location, visibility)
+		return mproperty
+	end
+
+	# Creation of a new method (AST and model representation) with the given name.
+	# See `create_method_from_property` for more information.
+	fun create_method_from_name(name: String, mclassdef: MClassDef, msignature: nullable MSignature, visibility: nullable MVisibility): AMethPropdef do
+		var mproperty = get_mmethod(name, mclassdef, visibility)
+		return create_method_from_property(mproperty, mclassdef, msignature)
+	end
+
+	# Same as `create_method_from_name` except the returned AMethPropdef is abstract.
+	# See `create_abstract_method_from_property` for more information.
+	fun create_abstract_method_from_name(name: String, mclassdef: MClassDef, msignature: nullable MSignature, visibility: nullable MVisibility): AMethPropdef do
+		var mproperty = get_mmethod(name, mclassdef, visibility)
+		return create_abstract_method_from_property(mproperty, mclassdef, msignature)
+	end
+
+	# Creation of a new method (AST and model representation) with the given MMethod.
+	# See `create_abstract_method_from_property` for more information.
+	# Take care, the AMethPropdef returned has an empty body (potential error if the given signature has an return type).
+	fun create_method_from_property(mproperty: MMethod,  mclassdef: MClassDef, msignature: nullable MSignature): AMethPropdef do
+		var n_def = create_abstract_method_from_property(mproperty, mclassdef, msignature)
+		n_def.mpropdef.is_abstract = false
+		n_def.n_block = ast_builder.make_block
+		return n_def
+	end
+
+	# Creation of a new abstract method (AST and model representation) with the given MMethod
+	fun create_abstract_method_from_property(mproperty: MMethod,  mclassdef: MClassDef, msignature: nullable MSignature): AMethPropdef do
+		# new methoddef definition
+		var m_def = new MMethodDef(mclassdef, mproperty, mclassdef.location)
+		if msignature == null then msignature = new MSignature(new Array[MParameter])
+		# set the signature
+		m_def.msignature = msignature
+		# set the abstract flag
+		m_def.is_abstract = true
+		# Build ast representation of the method
+		var n_def = m_def.create_ast_representation(ast_builder)
+		# Association new npropdef to mpropdef
+		toolcontext.modelbuilder.unsafe_add_mpropdef2npropdef(m_def,n_def)
+		return n_def
+	end
+
+	# Creation of a new attribute (AST and model representation) with the given name.
+	# See `create_attribute_from_property` for more information.
+	fun create_attribute_from_name(name: String, mclassdef: MClassDef, mtype: MType, visibility: nullable MVisibility, is_writable: nullable Bool, default_value: nullable AExpr): AAttrPropdef do
+		if visibility == null then visibility = public_visibility
+		# Try to get the attribut.
```

Comment:
```suggestion
		# Try to get the attribute.
```

But honestly, since the method is called `try_get_...` and the variable `mattribute`, I guessed that part. A good comment would be explaining what happens if the try fails.